src/components/Link.react.js

```diff
@@ -2,69 +2,72 @@
 
 import PropTypes from 'prop-types';
 
-import React, {Component} from 'react';
+import React, { Component } from 'react';
 
 /*
  * event polyfill for IE
  * https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent
  */
-function CustomEvent (event, params) {
-    params = params || {
-        bubbles: false,
-        cancelable: false,
-        detail: undefined
-    };
-    var evt = document.createEvent('CustomEvent');
-    evt.initCustomEvent(
-        event, params.bubbles, params.cancelable, params.detail);
-    return evt;
+function CustomEvent(event, params) {
+  params = params || {
+    bubbles: false,
+    cancelable: false,
+    detail: undefined
+  };
+  var evt = document.createEvent('CustomEvent');
+  evt.initCustomEvent(event, params.bubbles, params.cancelable, params.detail);
+  return evt;
 }
 CustomEvent.prototype = window.Event.prototype;
 
 export default class Link extends Component {
-    constructor(props) {
-        super(props);
-        this.updateLocation = this.updateLocation.bind(this);
-    }
+  constructor(props) {
+    super(props);
+    this.updateLocation = this.updateLocation.bind(this);
+  }
 
-    updateLocation() {
-        const {href, refresh} = this.props;
-        if (refresh) {
-            window.location.pathname = href;
-        } else {
-            window.history.pushState({}, '', href);
-            window.dispatchEvent(new CustomEvent('onpushstate'));
-        }
+  updateLocation(e) {
+    e.preventDefault(); // prevent anchor from updating location
+    const { href, refresh } = this.props;
```

Comment:
Similarly here, our convention is no white spaces around `{}`, so `const {href, refresh} = this.props`